core/src/main/scala/org/apache/spark/scheduler/Stage.scala

```diff
@@ -110,3 +133,8 @@ private[spark] abstract class Stage(
     case _ => false
   }
 }
+
+private[scheduler] object Stage {
+  // The number of consecutive failures allowed before a stage is aborted
+  val MAX_CONSECUTIVE_FAILURES = 4
```

Comment:
to clarify a little bit -- fetch failures are the *only* way we currently fail stages.  Separately there are task failures, and job failures.  In any case its good to make that clear here.